builder/pipeline.cc

```diff
@@ -711,6 +712,21 @@ class pipeline_builder {
         }
       }
     }
+
+    // Check to see if there are any *intermediate* outputs that don't
+    // have allocation bounds; if there are, create an empty allocation
+    // bounds for them.
+    for (const func* f : order_) {
+      for (const auto& o : f->outputs()) {
+        if (output_syms_.count(o.sym())) continue;
+        if (allocation_bounds_[o.sym()]) continue;
+        box_expr crop(o.buffer->rank());
+        for (std::size_t d = 0; d < crop.size(); ++d) {
+          crop[d] = {0, 0};
```

Comment:
This isn't empty, it's {min, max}.

In order to make this work, I think this needs to be `{std::numeric_limits<index_t>::max(), std::numeric_limits<index_t>::min()}`. We don't just want an empty dimension, we want the identity dimension for the union operator.

Reply:
Actually, we have a name for this `interval_expr::union_identity()`: https://github.com/dsharlet/slinky/blob/2271ddacc38e6708e25f4513f0524fc6e18811a7/runtime/expr.h#L298

Reply:
> This isn't empty, it's {min, max}.

Yeah, I was thinking like Halide's {min, extent}.

What about {0, -1}?

Reply:
```
  * frame #4: 0x000000010009cb4b pipeline`slinky::dim::extent(this=0x00007ff7bfefcbc0) const at buffer.h:69:5
    frame #5: 0x00000001002cece3 pipeline`slinky::(anonymous namespace)::alloc_extent(dim=0x00007ff7bfefcbc0) at buffer.cc:30:16
    frame #6: 0x00000001002cea37 pipeline`slinky::raw_buffer::init_strides(this=0x00007ff7bfefcc98, alignment=1) at buffer.cc:135:30
    frame #7: 0x00000001002cee3e pipeline`slinky::raw_buffer::allocate(this=0x00007ff7bfefcc98) at buffer.cc:170:3
```

Reply:
Put `dim.max() >= dim.min() ? dim.extent() : 0` before that call to extent.

Reply:
PTAL